hazelcast/src/main/java/com/hazelcast/internal/metrics/ProbeUnit.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.metrics;
+
+/**
+ * Measurement unit of a Probe. Not used on member, becomes a part of the key.
+ */
+public enum ProbeUnit {
```

Comment:
Mind that these are extended/changed later. For that reason they should have a numeric ID that is not determined by their `ordinal`.

Reply:
why not use string names? So you allow for a lot of flexibility and give up on type safety.

Reply:
strings essentially means its a contract based on convention which I dislike. Its clumsy stringly typed programming we already have enough of. I want actual semantics.